test/clippy/clippy_failure_test.sh

```diff
@@ -30,6 +33,20 @@ function check_build_result() {
     echo "FAIL: Unexpected return code [saw: ${ret}, want: ${1}] building target //test/clippy:${2}"
     echo "  Run \"bazel build //test/clippy:${2}\" to see the output"
     exit 1
+  elif [[ $# -ge 3 ]] && [[ "${@:3}" == *"capture_clippy_output"* ]]; then
+    # Make sure that content was written to the output file
+    if [ "$(uname)" == "Darwin" ]; then
+      STATOPTS=(-f%z)
+    else
+      STATOPTS=(-c%s)
+    fi
+    if [[ $(stat ${STATOPTS[@]} "${NEW_WORKSPACE}/bazel-bin/test/clippy/${2%_clippy}.clippy.out") == 0 ]]; then
+      echo "FAIL: Output wasn't written to out file building target //test/clippy:${2}"
```

Comment:
Nit: Maybe also print the filename in the error message. Although, since it's in a temp dir, there might not be much point since it will be deleted.